itest/lnd_quiescence_test.go

```diff
@@ -0,0 +1,32 @@
+package itest
+
+import (
+	"github.com/btcsuite/btcd/btcutil"
+	"github.com/lightningnetwork/lnd/lnrpc"
+	"github.com/lightningnetwork/lnd/lntest"
+	"github.com/stretchr/testify/require"
+)
+
+// testQuiescence tests whether we can come to agreement on quiescence of a
+// channel. We initiate quiescence via RPC and if it succeeds we verify that
+// the expected initiator is the resulting initiator.
+//
+// NOTE FOR REVIEW: this could be improved by blasting the channel with HTLC
+// traffic on both sides to increase the surface area of the change under test.
+func testQuiescence(ht *lntest.HarnessTest) {
+	alice, bob := ht.Alice, ht.Bob
+
+	chanPoint := ht.OpenChannel(bob, alice, lntest.OpenChannelParams{
+		Amt: btcutil.Amount(1000000),
+	})
+	defer ht.CloseChannel(bob, chanPoint)
+
+	ht.AssertTopologyChannelOpen(bob, chanPoint)
+
+	res, err := alice.RPC.Quiesce(&lnrpc.QuiescenceRequest{
+		ChanId: chanPoint,
+	})
+
```

Comment:
We can use the `hodl` HTLC flags (see some of the force close itests) to create an HTLC that isn't settled by remote party immediately, then manually settle that and assert that the stfu finishes. We used a similar pattern for the updated co-op close itests.

Reply:
The condition for allowing quiescence isn't about whether the htlcs have settled, but rather whether state is bilaterally consistent. We don't presently have hooks for that nor do we have a means of witholding commitment transactions either.